[CI] InternalExtendedStatsTests.testReduceRandom fails

InternalExtendedStatsTests.testReduceRandom fails with the following error:

<details>

```
   [junit4] <JUnit4> says नमस्ते! Master seed: BB928DDDA878E679
==> Test Info: seed=BB928DDDA878E679; jvm=1; suite=1
Suite: org.elasticsearch.search.aggregations.metrics.InternalExtendedStatsTests
  1> [2017-05-29T10:16:42,930][WARN ][o.e.b.JNANatives         ] Unable to lock JVM Memory: error=12, reason=Ne peut allouer de la mémoire
  1> [2017-05-29T10:16:42,933][WARN ][o.e.b.JNANatives         ] This can result in part of the JVM being swapped out.
  1> [2017-05-29T10:16:42,933][WARN ][o.e.b.JNANatives         ] Increase RLIMIT_MEMLOCK, soft limit: 65536, hard limit: 65536
  1> [2017-05-29T10:16:42,933][WARN ][o.e.b.JNANatives         ] These can be adjusted by modifying /etc/security/limits.conf, for example: 
  1>    # allow user 'tanguy' mlockall
  1>    tanguy soft memlock unlimited
  1>    tanguy hard memlock unlimited
  1> [2017-05-29T10:16:42,934][WARN ][o.e.b.JNANatives         ] If you are logged in interactively, you will have to re-login for the new limits to take effect.
  1> [2017-05-29T04:16:43,823][INFO ][o.e.s.a.m.InternalExtendedStatsTests] [testReduceRandom]: before test
  1> [2017-05-29T04:16:43,891][INFO ][o.e.s.a.m.InternalExtendedStatsTests] [testReduceRandom]: after test
  2> REPRODUCE WITH: gradle :core:test -Dtests.seed=BB928DDDA878E679 -Dtests.class=org.elasticsearch.search.aggregations.metrics.InternalExtendedStatsTests -Dtests.method="testReduceRandom" -Dtests.security.manager=true -Dtests.jvm.argline="-XX:-UseConcMarkSweepGC -XX:+UseG1GC" -Dtests.locale=th -Dtests.timezone=America/Manaus
FAILURE 0.53s | InternalExtendedStatsTests.testReduceRandom <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: expected:<8.915190858759545E7> but was:<8.915190858759534E7>
   >    at __randomizedtesting.SeedInfo.seed([BB928DDDA878E679:7BAF21CC8E257F2F]:0)
   >    at org.elasticsearch.search.aggregations.metrics.InternalExtendedStatsTests.assertReduced(InternalExtendedStatsTests.java:85)
   >    at org.elasticsearch.search.aggregations.metrics.InternalExtendedStatsTests.assertReduced(InternalExtendedStatsTests.java:34)
   >    at org.elasticsearch.test.InternalAggregationTestCase.testReduceRandom(InternalAggregationTestCase.java:245)
   >    at java.lang.Thread.run(Thread.java:745)
  2> NOTE: leaving temporary files on disk at: /home/tanguy/Github/elasticsearch/core/build/testrun/test/J0/temp/org.elasticsearch.search.aggregations.metrics.InternalExtendedStatsTests_BB928DDDA878E679-001
  2> NOTE: test params are: codec=Asserting(Lucene70): {}, docValues:{}, maxPointsInLeafNode=1948, maxMBSortInHeap=7.81230414099695, sim=RandomSimilarity(queryNorm=true): {}, locale=th, timezone=America/Manaus
  2> NOTE: Linux 4.4.0-78-generic amd64/Oracle Corporation 1.8.0_121 (64-bit)/cpus=8,threads=1,free=497547272,total=536870912
  2> NOTE: All tests run in this JVM: [InternalExtendedStatsTests]
Completed [1/1] in 1.40s, 1 test, 1 failure <<< FAILURES!

Tests with failures:
  - org.elasticsearch.search.aggregations.metrics.InternalExtendedStatsTests.testReduceRandom
```
</details>

The failure can be reproduced using:
```
gradle :core:test -Dtests.seed=BB928DDDA878E679 -Dtests.class=org.elasticsearch.search.aggregations.metrics.InternalExtendedStatsTests -Dtests.method="testReduceRandom" -Dtests.security.manager=true -Dtests.jvm.argline="-XX:-UseConcMarkSweepGC -XX:+UseG1GC" -Dtests.locale=th -Dtests.timezone=America/Manaus
```

It failed today on CI, see https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/2981/console

--- a/core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStatsTests.java
+++ b/core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStatsTests.java
@@ -79,10 +79,13 @@ public class InternalExtendedStatsTests extends InternalAggregationTestCase<Inte
         }
         assertEquals(sigma, reduced.getSigma(), 0);
         assertEquals(expectedCount, reduced.getCount());
-        assertEquals(expectedSum, reduced.getSum(), 1e-07);
+        // The order in which you add double values in java can give different results. The difference can
+        // be larger for large sum values, so we make the delta in the assertion depend on the values magnitude
+        assertEquals(expectedSum, reduced.getSum(), Math.abs(expectedSum) * 1e-12);
         assertEquals(expectedMin, reduced.getMin(), 0d);
         assertEquals(expectedMax, reduced.getMax(), 0d);
-        assertEquals(expectedSumOfSquare, reduced.getSumOfSquares(), 1e-07);
+        // summing squared values, see reason for delta above
+        assertEquals(expectedSumOfSquare, reduced.getSumOfSquares(), expectedSumOfSquare * 1e-14);
     }
 
     @Override

